### ISSUE TITLE
[feature requesr] mark above/below read/unread

### ISSUE THREAD

**Issue by pcause** [martinrotter/rssguard#244]
A useful feature would be to be able to mark all items in the message list above/below a selected item as read or unread.  If one isn't marking read on scroll I could get to a place and want to have all previous items in the list marked read or unread and then also be able to reverse that.

A good companion feature is to be able to select an item and say keep read/unread".

**Comment by martinrotter** [comment]
Note, that RSS Guard already has actions "Go to next/previous (unread) message". Just a side note.

Will think about this. I mean, it can be relatively easily added. Just. You basically mean this:
1. Select some feed.
2. It's messages are displayed in messages list and they are (usually) sorted according to some column, let's say the are now sorted with "Title" column.
3. We have 10 messages in the list and they are all unread.
4. User clicks on 5. message in the list, it gets instantly marked as read, its contents are displayed in previewer.
5. ** And at this point all 4 messages which are placed above our 5. message get automatically marked as read too.

Do I get it right? How should user trigger the feature? Keyboard shortcut (+ menu item?) or automatically (if enabled in settings)? How would user decide if he wants to apply the feature for above or below messages?

**Comment by pcause** [comment]
Nope.  I click message 5, it get marked as read.  I right click (or bind a keystroke) and selected from the menu "Mark Above Read" and messages 1-4 are marked as read.  If I select "Mark below read" messages 6-end are marked read, but 1-4 are still unread.

**Comment by martinrotter** [comment]
Yes, now it makes completely sense and it is implementable. Will probably add this. :)

**Comment by pcause** [comment]
rhanks

**Comment by martinrotter** [comment]
It took just 5 years. Do you even still use RSS Guard?

**Comment by nemonein** [comment]
Hello.
I'm on 5.1.0 (Archlinux/KDE, qt6 version), I still do not see the menu. 'Mark above as read' or something.


<img width="630" height="623" alt="Image" src="https://github.com/user-attachments/assets/2c56cbd6-ac07-4d9a-ba48-0de29612df19" />

**Comment by martinrotter** [comment]
<img width="1298" height="830" alt="Image" src="https://github.com/user-attachments/assets/a22e8fb7-c67a-44ed-9b58-ec70de592dec" />

**Comment by nemonein** [comment]
Thank you for your answer.
Yes, there it is...
I wish I could do that with the context menu. I mean right click menu.
Anyway, I really appreciate it. Now it's time for me to ditch Inoreader, finally.

**Comment by martinrotter** [comment]
Well, the context menu would be long with it. Let's just keep it this way, you can of course assign key shortcuts in the settings.

**Comment by nemonein** [comment]
Ok, I see. Short is always good.
